File: ssafy_task/samsung_B/커피점제과점_Main.java
package ssafy_task.samsung_B;

import java.util.Scanner;

class 커피점제과점_Main {
	private final static int MAX_E = 30000;
	private final static int MAX_SHOP = 1000;
	private final static int CMD_INIT = 100;
	private final static int CMD_ADD = 200;
	private final static int CMD_CALC = 300;

	private final static 커피점제과점_UserSolution usersolution = new 커피점제과점_UserSolution();

	private static boolean run(Scanner sc) {
		int q = sc.nextInt();

		int n, k, m, p, r;
		int[] sBuildingArr = new int[MAX_E];
		int[] eBuildingArr = new int[MAX_E];
		int[] mDistArr = new int[MAX_E];
		int[] mCoffee = new int[MAX_SHOP];
		int[] mBakery = new int[MAX_SHOP];
		int sBuilding, eBuilding, mDist;
		int cmd, ans, ret = 0;
		boolean okay = false;

		for (int i = 0; i < q; ++i) {
			cmd = sc.nextInt();
			switch (cmd) {
				case CMD_INIT:
					okay = true;
					n = sc.nextInt();
					k = sc.nextInt();
					for (int j = 0; j < MAX_E; ++j) {
						sBuildingArr[j] = -1;
						eBuildingArr[j] = -1;
						mDistArr[j] = 0;
					}
					for (int j = 0; j < k; ++j) {
						sBuildingArr[j] = sc.nextInt();
						eBuildingArr[j] = sc.nextInt();
						mDistArr[j] = sc.nextInt();
					}
					usersolution.init(n, k, sBuildingArr, eBuildingArr, mDistArr);
					break;
				case CMD_ADD:
					sBuilding = sc.nextInt();
					eBuilding = sc.nextInt();
					mDist = sc.nextInt();
					usersolution.add(sBuilding, eBuilding, mDist);
					break;
				case CMD_CALC:
					m = sc.nextInt();
					p = sc.nextInt();
					r = sc.nextInt();
					for (int j = 0; j < MAX_SHOP; ++j) {
						mCoffee[j] = -1;
						mBakery[j] = -1;
					}
					for (int j = 0; j < m; ++j) {
						mCoffee[j] = sc.nextInt();
					}
					for (int j = 0; j < p; ++j) {
						mBakery[j] = sc.nextInt();
					}
					ret = usersolution.calculate(m, mCoffee, p, mBakery, r);
					ans = sc.nextInt();
					if (ans != ret)
						okay =false;
					break;
				default:
					okay = false;
					break;
			}
		}
		return okay;
	}

	public static void main(String[] args) throws Exception {
		int TC, MARK;

		//System.setIn(new java.io.FileInputStream("res/sample_input.txt"));

		Scanner sc = new Scanner(System.in);

		TC = sc.nextInt();
		MARK = sc.nextInt();

		for (int testcase = 1; testcase <= TC; ++testcase) {
			int score = run(sc) ? MARK : 0;
			System.out.println("#" + testcase + " " + score);
		}

		sc.close();
	}
}

File: ssafy_task/samsung_B/커피점제과점_main.py
import sys

from 커피점제과점_solution_가지치기코드 import init, add, calculate

CMD_INIT = 100
CMD_ADD = 200
CMD_CALC = 300

def run():
	q = int(sys.stdin.readline())
	okay = False

	sBuildingArr = []
	eBuildingArr = []
	mDistArr = []

	for i in range(q):
		cmd = int(sys.stdin.readline())

		if cmd == CMD_INIT:
			inputarray = sys.stdin.readline().split()
			n = int(inputarray[0])
			k = int(inputarray[1])
			for _ in range(k):
				road = sys.stdin.readline().split()
				sBuildingArr.append(int(road[0]))
				eBuildingArr.append(int(road[1]))
				mDistArr.append(int(road[2]))

			init(n, k, sBuildingArr, eBuildingArr, mDistArr)
			okay = True
		elif cmd == CMD_ADD:
			inputarray = sys.stdin.readline().split()
			sBuilding = int(inputarray[0])
			eBuilding = int(inputarray[1])
			mDist = int(inputarray[2])
			add(sBuilding, eBuilding, mDist)
		elif cmd == CMD_CALC:
			inputarray = sys.stdin.readline().split()
			m = int(inputarray[0])
			p = int(inputarray[1])
			r = int(inputarray[2])
			mCoffee = []
			for _ in range(m):
				mCoffee.append(int(sys.stdin.readline()))
			mBakery = []
			for _ in range(p):
				mBakery.append(int(sys.stdin.readline()))

			ans = int(sys.stdin.readline())
			ret = calculate(m, mCoffee, p, mBakery, r)
			if ans != ret:
				okay = False
		else:
			okay = False

	return okay


if __name__ == '__main__':
	#sys.stdin = open('sample_input.txt', 'r')
	inputarray = sys.stdin.readline().split()
	TC = int(inputarray[0])
	MARK = int(inputarray[1])

	for testcase in range(1, TC + 1):
		score = MARK if run() else 0
		print("#%d %d" % (testcase, score), flush = True)